### ISSUE TITLE
an expected outcome for each test

### ISSUE THREAD

**Issue by ent0n29** [ent0n29/compilers-assignments#9]
Since we have our IR code to optimize with our passes, we don't know if our optimized .ll versions are correct.
We need a correct expected outcome for each test, and to find a way to obtain these expected results.

**Comment by ent0n29** [comment]
### suggestion for testing
check diffs between optimized and expected outcome
```bash
$ diff <(cat name_of_file_optimized.ll) <(cat name_of_expected_outcome.ll)
```

**Comment by ent0n29** [comment]
done with e73a393230410034378b7c9e968bf0dd9284dc79 by @ch-fano, currently refactoring folders for testing for better readability.

**Comment by ent0n29** [comment]
opt generated 2 unwanted lines in the _optimized.ll code, fixed the diff command with 2b625b2a11897422e5be77321d911cfc4a4cf956

**Comment by ent0n29** [comment]
everything fixed, only some refactoring and better readability needed
